### PR TITLE
update API updateStatus for devices

### PR DESCRIPTION
**Pull request by huyle0110**

### CODE REVIEW

File: devices/updateStatus.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      return;
+    }
+
+    if (result.Item.chipId != input.chipId) {
+      console.error(error);
+      callback(null, utils.createResponse(500, 'Invalid chipId '));
+      return;
+    }
+
+    if (userRole == constants.ROLE_BROKER) {
```
Comment (vanminh0910) [inline devices/updateStatus.js]: This should be checked at the beginning, and should return 403 forbidden, not 500 internal server error

File: devices/updateStatus.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      var attributeUpdates = utils.toAttributeUpdates({
+        status: 0,
```
Comment (vanminh0910) [inline devices/updateStatus.js]: use constants.STATUS_ONLINE or constants.STATUS_OFFLINE, why is it always 0 here?

Comment (huyle0110) [reply devices/updateStatus.js]: oh i dont know both  , will use both now

File: devices/updateStatus.js (hunk 1 above)
Comment (huyle0110) [reply devices/updateStatus.js]: i though the fist this API should check device not exist or exist   , then will update

File: devices/update.js [hunk 3]
```diff
@@ -40,6 +40,24 @@ module.exports.update = (event, context, callback) => {
     });
 
     dynamodb.update({
+<<<<<<< HEAD
```
Comment (vanminh0910) [inline devices/update.js]: This is conflict with master branch. Please pull latest code from master to your branch

File: devices/updateStatus.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    TableName: process.env.DEVICES_TABLE_NAME,
+    IndexName: 'DevicesChipIdIndex',
+    KeyConditionExpression: 'chipId = :chipId',
+    ExpressionAttributeValues: { ':chipId': input.chipId }
+  };
+
+  dynamodb.query(params, function (err, result) {
+    if (err) {
+      console.log(err);
+      callback(null, utils.createResponse(500, 'An internal server error occurred'));
```
Comment (vanminh0910) [inline devices/updateStatus.js]: Change to 

callback(null, utils.createResponse(500));

File: devices/updateStatus.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const dynamodb = require('../lib/dynamodb');
+const utils = require('../lib/utils');
+var _ = require('underscore');
+const constants = require('../lib/constants');
+
+module.exports.updateStatus = (event, context, callback) => {
+
+  const input = JSON.parse(event.body);
```
Comment (vanminh0910) [inline devices/updateStatus.js:10]: Need to validate input here. The chipId and status must not be blank, status must be online or offline value. Return 400 Invalid device data if data error found

File: devices/updateStatus.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  };
+
+  dynamodb.query(params, function (err, result) {
+    if (err) {
+      console.log(err);
+      callback(null, utils.createResponse(500, 'An internal server error occurred'));
+      return;
+    }
+    var attributeUpdates = utils.toAttributeUpdates({
+      status: constants.STATUS_OFFLINE,
```
Comment (vanminh0910) [inline devices/updateStatus.js]: status should use input.status, why is it always offline here?

File: devices/updateStatus.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    ExpressionAttributeValues: { ':chipId': input.chipId }
+  };
+
+  dynamodb.query(params, function (err, result) {
+    if (err) {
+      console.log(err);
+      callback(null, utils.createResponse(500, 'An internal server error occurred'));
+      return;
+    }
+    var attributeUpdates = utils.toAttributeUpdates({
```
Comment (vanminh0910) [inline devices/updateStatus.js]: Need to check if you can find the device with given chip Id or not before updating

File: devices/updateStatus.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      status: constants.STATUS_OFFLINE,
+      updatedAt: new Date().getTime()
+    });
+    var result = JSON.stringify(result.Items);
```
Comment (vanminh0910) [inline devices/updateStatus.js]: I don't think we need stringify here, result.Items should be an json array already. Also need to check if the result.length = = 1 or not. It should be 1 or raise error.

When length is 1, you can use result.Items[0], no need slicing

File: lib/brokerAuth.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
+'use strict';
```
Comment (vanminh0910) [inline lib/brokerAuth.js]: If we use API token, we don't need this auth handler

File: lib/utils.js [hunk 10]
```diff
@@ -49,7 +49,10 @@ var utils = {
 			expiresIn: '2 days'
 		});
 	},
-
+    decodedToken : function (token) {
```
Comment (vanminh0910) [inline lib/utils.js]: The indentation is not correct

File: lib/utils.js [hunk 11]
```diff
@@ -49,7 +49,10 @@ var utils = {
...
+	  return jwt.verify(token,config.jwt.secret);
```
Comment (vanminh0910) [inline lib/utils.js]: should add space after ,

File: lib/utils.js [hunk 12]
```diff
@@ -49,7 +49,10 @@ var utils = {
...
+	 
+	},
 	toAttributeUpdates: function (keyValues) {
```
Comment (vanminh0910) [inline lib/utils.js]: This does not have latest code from master

File: devices/update.js (hunk 3 above)
Comment (huyle0110) [reply devices/update.js]: fixed

File: devices/updateStatus.js (hunk 4 above)
Comment (huyle0110) [reply devices/updateStatus.js]: fixed

File: devices/updateStatus.js (hunk 6 above)
Comment (huyle0110) [reply devices/updateStatus.js]: oh i though when broker call this API  , it will be updated to offline . 
i am fixing

**Pull request closed without merging**